xcube/core/subsampling.py

```diff
@@ -109,6 +113,10 @@ def subsample_dataset(
     return xr.Dataset(data_vars=new_data_vars, attrs=dataset.attrs)
 
 
+def _mode(x, axis, **kwargs):
+    return stats.mode(x, axis, nan_policy="omit", **kwargs).mode
```

Comment:
This eagerly loads all data into memory, even for dask arrays. Please switch to dask version if `x` is a dask array.